packages/venia-ui/lib/components/CartPage/ProductListing/product.js

```diff
@@ -95,7 +92,9 @@ export const REMOVE_ITEM_MUTATION = gql`
         removeItemFromCart(input: { cart_id: $cartId, cart_item_id: $itemId }) {
             cart {
                 id
```

Comment:
Don't need this id stub anymore since you're using the fragment.

Reply:
So fragments should always include it? This was one of those things I wasn't sure about. Do we just always make sure to have some identifier for caching in our queries/mutations or can we assume that fragment authors will include it?

Reply:
Yes, I think we should make this assumption that all fragments include id when available. Ad-hoc inclusion of id would imply we're relying on heuristic fragment matching, which requires exact field matches, which is difficult to maintain. It's not causing any harm, but could lead to confusion that you always need to add an id field, when you don't.

Reply:
Do you still feel this way? I kind of prefer that we let fragments only define the data they need.

Reply:
I don't think fragments should need any additional config besides inclusion to work; without id in the fragment, developers would need to be aware they need it for cache updates to work.

Reply:
It makes sense to me that the base "wrapping" query would include the `id` field, not the fragments.

Reply:
This seems like it could go either way -- either way we need to include `id` somewhere and it does not hurt to declare it multiple times so I'll just leave it as is.

Reply:
Great compromise 🤝